### PR TITLE
fix: Document why we  delay API responses by default and slightly improve logic

### PR DESCRIPTION
**Pull request by untitaker**
Not really sure why this was set to non-0 in the first place.

### CODE REVIEW

**Review by dashed — APPROVED** [review]

**Comment by dcramer** [comment]
I'm not a fan of removing this. The reason it's here is because it simulates a small amount of lag which helps uncover more obvious race conditions in UI interactions. It's also needed to test (or implement) any kind of loading scenarios. Without this we will just implicitly lower the overall quality of software we ship because we will not experience it in the same way we would in production.

**Comment by EvanPurkhiser** [comment]
Maybe we could add what @dcramer said as a comment.

**Comment by leedongwei** [comment]
I've played with this from 0-30000. IMO 150ms doesn't affect the developer QoL in a noticeable way. I had a wild thought during the enterprise perf project to increase it 10000 so everyone could experience some pain of our customers.

**Comment by untitaker** [comment]
Race conditions are probably as much hidden by this as much as others are revealed while some of our less great UI response times that we have in prod right now are only revealed by working on a realistic amount of data.

My local devenv is already slower than production and I'm working on neither frontend nor endpoints most of the time, so I'm just going to set this to 0 for myself.

**Comment by dcramer** [comment]
@untitaker the race this easy catches is when you are not waiting for data in a component render cycle. without this its possible for the api to respond faster than the render cycle. 150ms shouldnt really be that noticeable unless the page is already slow (doing too many queries). if we want to reduce it some we can, i just picked numbers a bit anecdotally, and because this has helped _so much_ in my past for identifying bugs, ive adopted this pattern in almost every project I build.

**Comment by untitaker** [comment]
cool, updated PR

**Review by dashed — APPROVED** [review]

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -1369,6 +1369,14 @@ def create_partitioned_queues(name):
 )
 
 # Delay (in ms) to induce on API responses
+#
+# Simulates a small amount of lag which helps uncover more obvious race
+# conditions in UI interactions. It's also needed to test (or implement) any
```
Comment (BYK) [inline src/sentry/conf/server.py:1374]: Shouldn't we add a random +/- difference to the base value if we want to find race conditions with this?

Comment (dcramer) [reply src/sentry/conf/server.py:1374]: It's not that kind of race - its more like, the UI renders and doesnt account for the fact that an API request might not return instantaneously. So you end up in a weirdly broken UI state that you don't notice in most situations. Variability already exists on the server, its just more about forcing _some_ kind of delay that is human noticeable.

**Pull request merged**